sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/JacksonAdapter.java

```diff
@@ -125,7 +125,10 @@ public JacksonAdapter() {
      *
      * @param configureSerialization Applies additional configuration to outer mapper using inner mapper for module
      * chaining.
+     * @deprecated This API will be removed in the future. Please use {@link #createDefaultSerializerAdapter()} if you
+     * need to use JacksonAdapter.
      */
+    @Deprecated
     public JacksonAdapter(BiConsumer<ObjectMapper, ObjectMapper> configureSerialization) {
```

Comment:
Deprecate the whole class too? Is this class going to remain at all?

Reply:
It should be removed later on but that is a much noisier deprecation given JacksonAdapter's usage in code generation